source/adios2/toolkit/sst/dp/rdma_dp.c

```diff
@@ -542,7 +543,21 @@ static void RdmaProvideWriterDataToReader(CP_Services Svcs,
                       rc);
     }
 
-    rc = drc_access(Fabric->credential, 0, &Fabric->drc_info);
+    // at large scale the servers backing drc can become overwhelmed and
+    // timeout. The return value is -22 (EINVAL). a work around is to wait
+    // and try again.
+    int try
+        = 0;
+    int max_try = 60;
+    int try_wait_sec = 1;
+    do
+    {
+        sleep(try_wait_sec);
+        rc = drc_access(Fabric->credential, 0, &Fabric->drc_info);
+        try
+            += 1;
+    } while ((rc != DRC_SUCCESS) && (try < max_try));
+
```

Comment:
The loop here always forces a sleep, even on the first try.  Maybe this could work?
```c++
rc = drc_access(Fabric->credential, 0, &Fabric->drc_info);
for(int retries = 0; (rc != DRC_SUCCESS) && (retries < max_retries); ++retries)
{
    usleep(retry_wait_usec);
    rc = drc_access(Fabric->credential, 0, &Fabric->drc_info);
}